client/cmd.go

```diff
@@ -215,14 +216,26 @@ func ReadTxCommandFlags(clientCtx Context, flagSet *pflag.FlagSet) (Context, err
 		clientCtx = clientCtx.WithSkipConfirmation(skipConfirm)
 	}
 
+	if clientCtx.SignModeStr == "" || flagSet.Changed(flags.FlagSignMode) {
+		signModeStr, _ := flagSet.GetString(flags.FlagSignMode)
+		clientCtx = clientCtx.WithSignModeStr(signModeStr)
+	}
+
 	if clientCtx.From == "" || flagSet.Changed(flags.FlagFrom) {
 		from, _ := flagSet.GetString(flags.FlagFrom)
-		fromAddr, fromName, err := GetFromFields(clientCtx.Keyring, from, clientCtx.GenerateOnly)
+		fromAddr, fromName, keyType, err := GetFromFields(clientCtx.Keyring, from, clientCtx.GenerateOnly)
 		if err != nil {
 			return clientCtx, err
 		}
 
 		clientCtx = clientCtx.WithFrom(from).WithFromAddress(fromAddr).WithFromName(fromName)
+
+		// If the `from` signer account is a ledger key, we need to use
+		// SIGN_MODE_AMINO_JSON, because ledger doesn't support proto yet.
+		// ref: https://github.com/cosmos/cosmos-sdk/issues/8109
+		if keyType == keyring.TypeLedger && clientCtx.SignModeStr != flags.SignModeLegacyAminoJSON {
```

Comment:
So should we add a warning message or not?

Reply:
Since it's trivial here I guess may be useful. Something like?

"default sign-mode 'direct' not supported by Ledger, using sign-mode 'amino-json'"

Reply:
Yes, I'm for adding a warning message.